### PR TITLE
Module code refactor

### PR DESCRIPTION
**Pull request by davidrapson**
~To avoid needing to preprocess lots of code in order run unit tests this PR reconfigures our tests to all run directly through mocha, replacing `phantomjs` for `jsdom` for the one test that requires a DOM (tabs). This PR also loads `jquery` as an npm dependency.~

The scope of this has expanded so reframing this:

- Removes all committed client-side vendor code (jQuery, Swiper, Grapnel) and loads them as npm dependencies instead
- Removes need for Grapnel and moves associated code into modules
- Decouples unit test code to not need to be precompiled, uses jsdom instead of phantomjs
- Removes need for global gulp in travis.yml and runs build step separately to tests

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -9,7 +9,7 @@ before_script:
 - npm install
 script:
 - npm run lint
-- gulp test --production
```
Comment (mattandrews) [inline .travis.yml]: This command does two things really, runs the tests but also compiles everything with the `--production` flag meaning that CSS/JS is minified etc. We'd need either add another step after this to compile assets, or add that flag to the `npm test` script.

File: assets/js/main.js [hunk 2]
```diff
@@ -1,12 +1,14 @@
...
 
+const $ = require('jquery');
+
 // initialise router (eg. run conditional code for certain URLs)
 const Grapnel = require('./libs/grapnel');
 const router = new Grapnel({ pushState: true });
 
 // initialise Vue
-const Vue = require('../../node_modules/vue/dist/vue.common');
+const Vue = require('vue');
```
Comment (mattandrews) [inline assets/js/main.js]: This is wrong – look in devtools and you'll see the warnings!

File: .travis.yml (hunk 1 above)
Comment (davidrapson) [reply .travis.yml]: Yep, as discussed, added that in as a separate step.

File: assets/js/main.js (hunk 2 above)
Comment (davidrapson) [reply assets/js/main.js]: Accidental hold-over from some webpack experimentation. Reinstated now.

**Pull request merged**